.github/workflows/build_and_test.yml

```diff
@@ -14,6 +14,28 @@ on:
         required: true
 
 jobs:
+  # This is on the top to give the most visibility in case of failures
+  hadoop-2:
+    name: Hadoop 2 build
+    runs-on: ubuntu-20.04
+    steps:
+    - name: Checkout Spark repository
+      uses: actions/checkout@v2
+    - name: Cache Coursier local repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.cache/coursier
+        key: hadoop-2-coursier-${{ hashFiles('**/pom.xml', '**/plugins.sbt') }}
+        restore-keys: |
+          hadoop-2-coursier-
+    - name: Install Java 11
+      uses: actions/setup-java@v1
+      with:
+        java-version: 11
```

Comment:
Does Java version cause any difference here?

Reply:
Actually, maybe we should stick to Java 8 because that's the Java version being used in the releases.

Reply:
No, @viirya . Apache Spark 3.0 already declared [Hadoop 2.7 + Hive 2.3 on JDK11](https://amplab.cs.berkeley.edu/jenkins/view/Spark%20QA%20Test%20(Dashboard)/job/spark-branch-3.0-test-sbt-hadoop-2.7-hive-2.3/).
> Does Java version cause any difference here?

Reply:
@HyukjinKwon . It will be okay~ This is only for compilation test for API instead of running real tests.

After we make `branch-3.1`, Jenkins will be created for all release profiles, too.

Reply:
Sure, I guess it's fine.